ddtrace/profiling/exporter/http.py

```diff
@@ -180,12 +158,20 @@ def export(self, events, start_time_ns, end_time_ns):
             # Retry after 1s, 2s, 4s, 8s with some randomness
             wait=tenacity.wait_random_exponential(multiplier=0.5),
             stop=tenacity.stop_after_delay(self.max_retry_delay),
-            retry=tenacity.retry_if_exception_type(
-                (error.HTTPError, error.URLError, http_client.HTTPException, OSError, IOError)
-            ),
+            retry=tenacity.retry_if_exception_type((http_client.HTTPException, OSError, IOError))
+            & tenacity.retry_if_exception(self._no_retry_on_404),
```

Comment:
IMO we should only be retrying on 5xx or connection failures since 4xx supposedly means something is permanently wrong with the request. 

Equivalent piece in dd-trace-go: https://github.com/DataDog/dd-trace-go/blob/v1/profiler/upload.go#L109

Reply:
Sure.